sys/include/net/nanocoap.h

```diff
@@ -2115,6 +2115,35 @@ ssize_t coap_payload_put_bytes(coap_pkt_t *pkt, const void *data, size_t len);
  */
 ssize_t coap_payload_put_char(coap_pkt_t *pkt, char c);
 
+/**
+ * @brief   Create CoAP reply header (convenience function)
+ *
+ * This function will generate the reply CoAP header and sets
+ * the payload pointer inside the response buffer to point to
+ * the start of the payload, so that it can be written directly
+ * after the header.
+ *
+ * @param[in]   pkt         packet to reply to
+ * @param[in]   code        reply code (e.g., COAP_CODE_204)
+ * @param[out]  buf         buffer to write reply to
+ * @param[in]   len         size of @p buf
+ * @param[in]   ct          content type of payload
+ *                          if ct < 0 this will be ignored
+ * @param[in]   payload     Will be set to the start of the payload inside
+ *                          @p buf.
+ *                          May be set to NULL if no payload response is
```

Comment:
That's actually just a special case that occurs with the no-response option.
If you already know you have no payload, there is no point in calling `coap_reply_simple()`

Reply:
What would you call instead if you just wanted to build an ACK frame? Maybe it would be good to mention that option then.

In general, I'm just pointing out things I notice, if you feel like it's out of the scope for this PR that's also fine with me.

Reply:
You can use `coap_build_reply_header()` if you want an ACK without payload.
For an empty ACK with a separate response, #20266 adds `coap_build_empty_ack()`